tests/test_utils.py

```diff
@@ -162,65 +162,85 @@ def check_if_job_finished(job: JobDataclass):
+        result: ResultDataclass = job.results[result_index]
+        check_standard_result_data(program_index, job, result)
+        assert result.meta is not None
         shots: int = job.shots
-        counts: dict = result.result_dict["counts"]
-        probabilities: dict = result.result_dict["probabilities"]
+        data: dict = result.data
 
         # Check if the first result is distributed correctly: 50% for the qubit zero and 50% for the qubit three
-        if (i % 2) == 0:
-            assert compare_values_with_tolerance(shots / 2, counts[BIT_0], COUNTS_TOLERANCE)
-            assert compare_values_with_tolerance(shots / 2, counts[BIT_3], COUNTS_TOLERANCE)
-            assert (counts[BIT_0] + counts[BIT_3]) == shots
-
-            assert compare_values_with_tolerance(PROBABILITY_1 / 2, probabilities[BIT_0], PROBABILITY_TOLERANCE)
-            assert compare_values_with_tolerance(PROBABILITY_1 / 2, probabilities[BIT_3], PROBABILITY_TOLERANCE)
-            assert (probabilities[BIT_0] + probabilities[BIT_3]) > PROBABILITY_1 - PROBABILITY_TOLERANCE
+        if result_index % 4 == 0:
+            assert result.result_type == ResultType.COUNTS
+            assert compare_values_with_tolerance(shots / 2, data[BIT_0], COUNTS_TOLERANCE)
+            assert compare_values_with_tolerance(shots / 2, data[BIT_3], COUNTS_TOLERANCE)
+            assert (data[BIT_0] + data[BIT_3]) == shots
+        elif result_index % 4 == 1:
+            assert result.result_type == ResultType.PROBABILITIES
+            assert compare_values_with_tolerance(PROBABILITY_1 / 2, data[BIT_0], PROBABILITY_TOLERANCE)
+            assert compare_values_with_tolerance(PROBABILITY_1 / 2, data[BIT_3], PROBABILITY_TOLERANCE)
+            assert (data[BIT_0] + data[BIT_3]) > PROBABILITY_1 - PROBABILITY_TOLERANCE
 
         # Check if the first result is distributed correctly: 100% for the qubit zero
-        else:
-            assert counts[BIT_0] == shots
-            assert probabilities[BIT_0] == PROBABILITY_1
+        elif result_index % 4 == 2:
+            assert result.result_type == ResultType.COUNTS
+            assert data[BIT_0] == shots
+        elif result_index % 4 == 3:
+            assert result.result_type == ResultType.PROBABILITIES
+            assert data[BIT_0] == PROBABILITY_1
 
 
 def compare_values_with_tolerance(value1, value2, tolerance) -> bool:
     return value1 + tolerance > value2 > value1 - tolerance
 
 
-def check_standard_result_data(i, job, result):
-    assert result.result_type == ResultType.get_result_type(job.type)
+def check_standard_result_data(program_index: int, job, result):
     assert result.result_type != ResultType.ERROR, result
     assert result.job_id == job.id
     if hasattr(result, "circuit"):  # result is DTO
-        assert result.circuit == job.deployment.programs[i].quantum_circuit
+        assert result.circuit == job.deployment.programs[program_index].quantum_circuit
```

Comment:
```suggestion
        assert result.circuit == db_result.program.quantum_circuit
```
Use the result object from the DB to test against in these asserts to avoid calculating the `program_index`. Same for next assert. (I am fine with using result index or result id to fetch the correct result from the db)